haystack/nodes/reader/farm.py

```diff
@@ -239,7 +240,27 @@ def _training_procedure(
             use_amp=use_amp,
         )
         # 4. Feed everything to the Trainer, which keeps care of growing our model and evaluates it from time to time
-        if teacher_model: # checks again if teacher model is passed as parameter, in that case assume model distillation is used
+        if tinybert:
```

Comment:
Maybe we could first check `if teacher_model` and then have a nested check for whether to use tinybert or not. The else branch of the `if teacher_model` is then the standard use case without distillation.